src/test/java/seedu/sellsavvy/model/order/DateTest.java

```diff
@@ -20,24 +22,52 @@ public void constructor_invalidDate_throwsIllegalArgumentException() {
     }
 
     @Test
-    public void isValidDate() {
+    public void isValidDateRegex() {
         // null date value
-        assertThrows(NullPointerException.class, () -> Date.isValidDate(null));
+        assertThrows(NullPointerException.class, () -> Date.isValidDateRegex(null));
+
+        // invalid date regex
+        assertFalse(Date.isValidDateRegex("")); // empty string
+        assertFalse(Date.isValidDateRegex(" ")); // spaces only
+        assertFalse(Date.isValidDateRegex("date")); // not a date
+        assertFalse(Date.isValidDateRegex("2-02-2002")); // invalid day format
```

Comment:
On top of single digit days, can a single digit month be included as well?
e.g. 02-2-2002

Reply:
Both single digit days and month are currently not supported. You can reformat to allow for formatting to d-m-yy but thats just QoL.

Reply:
Updated in commit.